### PR TITLE
Update template

### PR DESCRIPTION
**Pull request by shiftsayan**
- Moved things around in the class for better structure
- Added Logic macros
- Left aligned the section headers so it was easy to modify them

### CODE REVIEW

File: homework/hw_example.tex [hunk 1]
```diff
@@ -12,7 +12,7 @@
 % Assignment Settings
 \newcommand{\homework}{3}
 
-\begin{document} 
+\begin{document}
```
Comment (thepinetree) [inline homework/hw_example.tex:15]: Remove unnecessary space

File: homework/plaid-homework.cls [hunk 2]
```diff
@@ -1,178 +1,195 @@
 %******************************************************************************%
-%*                                Header Info                                 *%
+%* Classes                                                                    *%
```
Comment (thepinetree) [inline homework/plaid-homework.cls:2]: Hmm I kinda like the centering, have been using this:
`echo "text" | boxes -a hcvc -s 80x3` to do it

File: homework/hw_example.tex [hunk 3]
```diff
@@ -54,6 +54,8 @@
 \problem{2}{Induction}
 Induction becomes so much easier with LaTeX! Take a look:
 
+\ul{hello}
```
Comment (thepinetree) [inline homework/hw_example.tex:57]: This doesn't build right now with ul

File: homework/plaid-homework.cls [hunk 4]
```diff
@@ -1,178 +1,195 @@
...
-\RequirePackage{centernot}
-% Package for math formatting (differentials)
-\RequirePackage{commath} 
-% Package for higher order integrals
-\RequirePackage{esint}
-
-% Package to easily display code
+% For better, more beautiful math support
+\RequirePackage{mathtools}
+% For better, more beautiful code support (including syntax coloring!)
```
Comment (thepinetree) [inline homework/plaid-homework.cls:34]: Newline before this comment for separation

File: homework/plaid-homework.cls [hunk 5]
```diff
@@ -1,178 +1,195 @@
...
-
-% Package to define page setup
-\RequirePackage[textwidth=16.5cm, textheight=23.5cm]{geometry}
-% Define side-margin size for odd pages
-\oddsidemargin0cm
+% For multi-line comments
+\RequirePackage{comment}
+% For easy placeholder text generation
+\RequirePackage{lipsum} 
+% For horizontally centering the \not command
```
Comment (thepinetree) [inline homework/plaid-homework.cls:82]: I think these go better with the math stuff above for separation of intent

File: homework/plaid-homework.cls [hunk 6]
```diff
@@ -1,178 +1,195 @@
...
-    linkcolor=red,    
-    filecolor=magenta,      
+    colorlinks=true,
+    filecolor=magenta,
     urlcolor=cyan,
 }
 
+% For image support
+\RequirePackage{graphicx}
+\graphicspath{{assets/}}
```
Comment (thepinetree) [inline homework/plaid-homework.cls:61]: Describe what this is in a comment

File: homework/plaid-homework.cls [hunk 7]
```diff
@@ -1,178 +1,195 @@
...
+% For creating figures and subfigures
+\RequirePackage{caption}
+\RequirePackage{subcaption}
+% For positioning figures
+\RequirePackage{float}
 
-% Package for multi-line comments
-\RequirePackage{comment} 
-% Package for latex conditionals
+% For conditionals
```
Comment (thepinetree) [inline homework/plaid-homework.cls:75]: Should mention this is for LaTeX imo to differentiate from code listings

File: homework/plaid-homework.cls [hunk 8]
```diff
@@ -1,178 +1,195 @@
...
+  captionpos=b,                      % sets the caption-position to bottom
+  breaklines=true,                   % sets automatic line breaking
+  breakatwhitespace=false,           % sets auto breaks only at whitespace
+  keywordstyle=\color{DarkMagenta},  % keyword coloring
+  commentstyle=\color{ForestGreen},  % comment coloring
+  stringstyle=\color{Maroon},        % string literal 
+  morecomment=[l][\color{RoyalBlue}]{\#} % more coloring for special strings
+}
 
+% SML Programming Declaration
```
Comment (thepinetree) [inline homework/plaid-homework.cls:166]: This has to come before the lstset if the language in the above command is changed to sml

File: homework/plaid-homework.cls [hunk 9]
```diff
@@ -181,93 +198,98 @@ basicstyle=\small\ttfamily,
...
+% Proofs 
+\newcommand{\im}{\item[$\implies$]}                 % implies as bullets
+\newcommand{\QED}{\hfill \qedsymbol}                % QED box
+\newcommand{\QEDtext}{\hfill \textsc{qed}}          % QED text
+\newcommand{\micdrop}{\QED}
+
+% Induction
+\newcommand{\pn}{Define P\left($n$\right) as: }
+\newcommand{\st}{\text{ s.t. }}                             
+\newcommand{\RHS}{\underline{\textsf{RHS:}}}
```
Comment (thepinetree) [inline homework/plaid-homework.cls:264]: Not a huge fan of the sf font, but not too attached to this either

File: homework/plaid-homework.cls [hunk 10]
```diff
@@ -181,93 +198,98 @@ basicstyle=\small\ttfamily,
...
-\newcommand{\E}{\mathbb{E}}         % Expectation 
-\newcommand{\V}{\mathbb{V}}         % Variance
-\newcommand*{\Comb}[2]{C^{#1}_{#2}} % Combinations
-% Distributed as
-% Style: A \distas {distribution} B
-\newcommand{\distas}[1]{\overset{#1}{\sim}} 
-
+% Usage: \eval {function} {var=lower bound} {upper bound}
+\renewcommand{\eval}[3]{\left.#1\right\rvert_{#2}^{#3}}
+% Usage: A \distas{distribution} B
```
Comment (thepinetree) [inline homework/plaid-homework.cls:278]: This should be in probability, not calculus

**Review by thepinetree — CHANGES_REQUESTED** [review]
Some minor nits and design decisions. Main thing is getting in induction template here so doc compiles.

File: homework/plaid-homework.cls (hunk 8 above)
Comment (shiftsayan) [reply homework/plaid-homework.cls:166]: ah I didn't know that!

File: homework/plaid-homework.cls (hunk 9 above)
Comment (shiftsayan) [reply homework/plaid-homework.cls:264]: Changed it to regular text. I am not a fan of texttt either, but I think just underline works.

File: homework/hw-example.tex [hunk 11]
```diff
@@ -1,23 +1,23 @@
...
 \newcommand{\homework}{3}
 
-\begin{document} 
+\begin{document}
 \passport{}
 \title
 
 \problem{1}{Math}
-LaTeX is awesome for math! Let's look evaluating the value of $m$ if $(m + 2) \times (3 + 4) = 21$ in two ways.
+\LaTeX is awesome for math! Let's look evaluating the value of $m$ if $(m + 2) \times (3 + 4) = 21$ in two ways.
```
Comment (thepinetree) [inline homework/hw-example.tex:20]: This needs a space after \LaTeX. Should be \LaTeX\ is

File: homework/hw-example.tex [hunk 12]
```diff
@@ -95,7 +116,7 @@
 
 \problem{3}{Programming}
 
-You can code in LaTeX too! Here's an intro to typesetting programs:
+You can code in \LaTeX too! Here's an intro to typesetting programs:
```
Comment (thepinetree) [inline homework/hw-example.tex:119]: Same as above

File: homework/hw-example.tex [hunk 13]
```diff
@@ -139,7 +160,7 @@
 \newpage
 
 \problem{4}{Graphing}
-Plotting in Latex is super fun! Here's a mini graph with some shading:
+Plotting in \LaTeX is super fun! Here's a mini graph with some shading:
```
Comment (thepinetree) [inline homework/hw-example.tex:163]: Same as above

**Review by thepinetree — CHANGES_REQUESTED** [review]
Just the spaces, then merge it.

**Review by thepinetree — APPROVED** [review]

**Pull request merged**